src/pycram/datastructures/world.py

```diff
@@ -321,20 +355,18 @@ def remove_object(self, obj: Object) -> None:
 
         :param obj: The object to be removed.
         """
+        self.object_lock.acquire()
         obj.detach_all()
 
-        self.objects.remove(obj)
+        if obj.name != "floor":
+            self.objects.remove(obj)
```

Comment:
Good question :laughing:

This is because the floor in the multiverse is the name of the empty scene, and removing it closes the server :sweat_smile:

Reply:
in #177 it is not only the floor that is not removed, it is any object with type environment :sweat_smile:

Reply:
I have now made this function return a boolean value indicating whether the simulator removed the object or not and if true the world would remove the object from its list of object else it will leave it be.